anyrun/src/main.rs

```diff
@@ -679,11 +766,17 @@ fn activate(app: &gtk::Application, runtime_data: Rc<RefCell<RuntimeData>>) {
                 main_vbox.add(&main_list);
                 main_list.show();
                 entry.grab_focus(); // Grab the focus so typing is immediately accepted by the entry box
+                entry.set_position(-1); // -1 moves cursor to end of text in case some text was restored
             }
 
-            if runtime_data.borrow().config.show_results_immediately {
-                // Get initial matches
-                refresh_matches(String::new(), runtime_data);
+            // Show initial results if state restoration is enabled or immediate results are configured
+            let should_show_results = {
+                let data = runtime_data.borrow();
+                data.config.persist_state || data.config.show_results_immediately
```

Comment:
This does not account for whether or not there is a valid persisted state, as in will still show results immediately even if there is no input which may not be what the user wants. Should check if a valid, non-expired state exists.